File: android/src/main/java/com/capacitorjs/plugins/googlemaps/GoogleMapGestures.kt
package com.capacitorjs.plugins.googlemaps

import org.json.JSONObject

class GoogleMapGestures(fromJSONObject: JSONObject) {
    var scroll: Boolean? = true
    var tilt: Boolean? = true
    var zoom: Boolean? = true
    var rotate: Boolean? = true

    init {
        if (fromJSONObject.has("scroll")) {
            scroll = fromJSONObject.getBoolean("scroll")
        }
        if (fromJSONObject.has("tilt")) {
            tilt = fromJSONObject.getBoolean("tilt")
        }
        if (fromJSONObject.has("rotate")) {
            rotate = fromJSONObject.getBoolean("rotate")
        }
        if (fromJSONObject.has("zoom")) {
            zoom = fromJSONObject.getBoolean("zoom")
        }
    }
}

File: ios/Plugin/GoogleMapGestures.swift
import Foundation
import Capacitor

public struct GoogleMapGestures {
    var scroll: Bool? = true
    var tilt: Bool? = true
    var zoom: Bool? = true
    var rotate: Bool? = true

    init(fromJSObject: JSObject) throws {
        scroll = fromJSObject["scroll"] as? Bool
        tilt = fromJSObject["tilt"] as? Bool
        zoom = fromJSObject["zoom"] as? Bool
        rotate = fromJSObject["rotate"] as? Bool
    }

}
